### PR TITLE
Josh/botbuilder tests

### LINKED ISSUE #4101: migrate builder.testing
#4039

### PR DESCRIPTION
**Pull request by Virtual-Josh**
Fixes #4101

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs [hunk 1]
```diff
@@ -26,12 +24,12 @@ public void BotStateSet_Properties()
 
             var stateSet = new BotStateSet(userState, convState);
 
-            Assert.AreEqual(stateSet.BotStates.Count, 2);
-            Assert.IsNotNull(stateSet.BotStates.OfType<UserState>().First());
-            Assert.IsNotNull(stateSet.BotStates.OfType<ConversationState>().First());
+            Assert.Equal(stateSet.BotStates.Count, 2);
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs:27]: >tateSet.BotStates.Count, 2) [](start = 26, length = 27)

The expected value (2) should be the first parameter: Assert.Equal(2, stateSet.BotStates.Count);

File: tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs [hunk 2]
```diff
@@ -48,12 +46,12 @@ public async Task BotStateSet_LoadAsync()
 
                 var stateSet = new BotStateSet(userState, convState);
 
-                Assert.AreEqual(stateSet.BotStates.Count, 2);
+                Assert.Equal(stateSet.BotStates.Count, 2);
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs:49]: >Assert.Equal(stateSet.BotStates.Count, 2) [](start = 16, length = 41)

Please reorder parameters everywhere.

File: tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs [hunk 3]
```diff
@@ -98,24 +96,24 @@ public async Task BotStateSet_ReturnsDefaultForNullValueType()
 
             var stateSet = new BotStateSet(userState, convState);
 
-            Assert.AreEqual(stateSet.BotStates.Count, 2);
+            Assert.Equal(stateSet.BotStates.Count, 2);
 
             var userObject = await userProperty.GetAsync(turnContext, () => null);
-            Assert.AreEqual(null, userObject);
+            Assert.Equal(null, userObject);
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs:102]: >Assert.Equa [](start = 12, length = 11)

Use Assert.Null(userObject);

File: tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs [hunk 4]
```diff
@@ -98,24 +96,24 @@ public async Task BotStateSet_ReturnsDefaultForNullValueType()
...
 
             // Ensure we also get null on second attempt
             userObject = await userProperty.GetAsync(turnContext, () => null);
-            Assert.AreEqual(null, userObject);
+            Assert.Equal(null, userObject);
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs:107]: Apply Assert.Null to all cases.

File: tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterOAuthTests.cs [hunk 5]
```diff
@@ -124,11 +123,11 @@ public async Task TokenResponsePropertiesEndPolling()
             await Task.Delay(3000);
 
             // Make sure it only polled once and it ended
-            Assert.AreEqual(1, callCount);
-            Assert.IsTrue(adapter.Logger.LogData.Contains("PollForTokenAsync completed without receiving a token"));
+            Assert.Equal(1, callCount);
+            Assert.True(adapter.Logger.LogData.Contains("PollForTokenAsync completed without receiving a token"));
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterOAuthTests.cs:127]: >Assert.True(adapter.Logger.LogData.Contains [](start = 12, length = 43)

Use Assert.Contains("PollForToken...", adapter.Logger.LogData);

**Comment by mrivera-ms** [comment]
>     [TestClass]

Did you mean to submit this file? It is not migrated.

---
Refers to: tests/Microsoft.Bot.Builder.Transcripts.Tests/CoreExtensionsTests.cs:12 in 6468c73. [](commit_id = 6468c734cfb6541c66ee04ae58fac7d2ed10b77e, deletion_comment = False)

**Comment by mrivera-ms** [comment]
>     [TestClass]

Not migrated.

---
Refers to: tests/Microsoft.Bot.Builder.Transcripts.Tests/CoreTests.cs:14 in 6468c73. [](commit_id = 6468c734cfb6541c66ee04ae58fac7d2ed10b77e, deletion_comment = False)

File: tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs (hunk 1 above)
Comment (Virtual-Josh) [reply tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs]: updated.

File: tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs (hunk 2 above)
Comment (Virtual-Josh) [reply tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs]: updated.

File: tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs (hunk 4 above)
Comment (Virtual-Josh) [reply tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs:107]: updated

File: tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterOAuthTests.cs (hunk 5 above)
Comment (Virtual-Josh) [reply tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterOAuthTests.cs]: updated

**Comment by Virtual-Josh** [comment]
> > ```
> > [TestClass]
> > ```
> 
> Did you mean to submit this file? It is not migrated.
> 
> Refers to: tests/Microsoft.Bot.Builder.Transcripts.Tests/CoreExtensionsTests.cs:12 in [6468c73](https://github.com/microsoft/botbuilder-dotnet/commit/6468c734cfb6541c66ee04ae58fac7d2ed10b77e). [](commit_id = [6468c73](https://github.com/microsoft/botbuilder-dotnet/commit/6468c734cfb6541c66ee04ae58fac7d2ed10b77e), deletion_comment = False)

Yes, I did. This file is used in the botbuilder test suite. Since I had to cut the TestContext parameter to make the tests pass. I'll work on updating the `Transcripts` tests next.

**Comment by Virtual-Josh** [comment]
> > ```
> > [TestClass]
> > ```
> 
> Not migrated.
> 
> Refers to: tests/Microsoft.Bot.Builder.Transcripts.Tests/CoreTests.cs:14 in [6468c73](https://github.com/microsoft/botbuilder-dotnet/commit/6468c734cfb6541c66ee04ae58fac7d2ed10b77e). [](commit_id = [6468c73](https://github.com/microsoft/botbuilder-dotnet/commit/6468c734cfb6541c66ee04ae58fac7d2ed10b77e), deletion_comment = False)

See explanation below.

**Pull request merged**